store/sqlstore/post_store.go

```diff
@@ -1313,11 +1313,18 @@ func (s *SqlPostStore) search(teamId string, userId string, params *model.Search
 
 		if params.OrTerms {
 			queryParams["Terms"] = "(" + strings.Join(strings.Fields(terms), " | ") + ")" + excludeClause
+
 		} else {
 			queryParams["Terms"] = "(" + strings.Join(strings.Fields(terms), " & ") + ")" + excludeClause
 		}
+		searchClause := ""
+
+		if !params.OrTerms && strings.Contains(terms, "_") {
+			searchClause = fmt.Sprintf("AND to_tsvector('english', replace(%s, '_', '/')) @@ plainto_tsquery('english', replace(:Terms, '_', '/'))", searchType)
```

Comment:
I am wondering what are the benefits that `plainto_tsquery` is giving us if we have to replace underscores with `/` anyways, which also works the same way for `to_tsquery`. And what if the search clause itself contains `/`? Wouldn't we need to escape that?

Reply:
it works in both options and if the search contains /  it returns the same results.  i believe checking if it contains underscores and the else statement is not needed if that's the case then. Let me run some test on it .

Reply:
`plainto_tsquery` is fine for handling the underscores, but it can't express boolean logic other than plain `AND`. I am wondering if we should just fall back on just casting to `::tsquery` and lose out on the language config part if a text contains underscores.

Reply:
Casting to ::tsquery works best if there will be only single search terms with underscores.In case of multiple terms with one having underscores it seems that is where the language config will be needed to cater for the stop words and text normalization.

Reply:
That is right. It's a tradeoff between expressing boolean logic versus text normalization. I was thinking that expressing the boolean logic is more important. But completely open to opinions on this one.

Wondering if @ethervoid or @mgdelacroix has any thoughts on this one.

Reply:
First of all thank you very much @antwigambrah for taking care of this, really appreciated :)

I agree with @agnivade my idea on this was to make a pre-normalization of the terms so we can use boolean operations as we've been doing. Changing the way we're returning results would result in complaints from users.

The idea is to provide extra features instead of different ones

Reply:
@ethervoid Then i believe checking if it contains underscores and direct casting will be the best option if that's the case .

Reply:
Yes, I agree.

Reply:
@antwigambrah yes that would be the option IMHO

Reply:
@agnivade @ethervoid Update provided